nixos/modules/programs/qgroundcontrol.nix

```diff
@@ -40,13 +44,23 @@ in
       '';
     };
 
-    # Security wrapper
-    security.wrappers.qgroundcontrol = {
-      source = lib.getExe cfg.package;
-      owner = "root"; # Sensible default; not setuid so this is not a security risk
-      group = "tty";
-      setgid = true;
+    # Use mkSecurityWrapper to set the real group ID to dialout to allow for unprivileged access to serial ports without causing GTK to crash
```

Comment:
This is a bit odd -- why exactly are you wrapping what seems to be a homemade setuid wrapper in another setuid wrapper?

Reply:
GTK has a check to ensure that the effective user and group ID equals the real user and group ID. This is a reasonable check, and is therefore not something I want to disable. However, it does mean I have to use a wrapper, since setgid only changes the effective user ID and not the real user ID.